File: frog_river_one/frog_river_one.js
// https://app.codility.com/demo/results/trainingJPUQ9P-26K/

function solution(X, A) {
  const set = new Set();
  let result = -1;

  for (let i = 0; i < A.length; i++) {
    set.add(A[i]);

    if (set.size >= X) {
      result = i;
      break;
    }
  }

  return result;
}

File: frog_river_one/frog_river_one.py
# https://app.codility.com/demo/results/trainingCCBPGX-AQW/


def solution(X, A):
    leaves_path = set([])
    result = -1

    for index, leave in enumerate(A):
        leaves_path.add(leave)

        if (len(leaves_path) == X):
            result = index
            break

    return result
